app/models/spree/payment.rb

```diff
@@ -126,7 +126,9 @@ def payment_source
     end
 
     def ensure_correct_adjustment
-      revoke_adjustment_eligibility if ['failed', 'invalid'].include?(state)
+      if ['failed', 'invalid'].include?(state) || authorization_action_required?
```

Comment:
The extra complexity this conditional has and I think it'd be good the extract it, give it an abstract name, but I don't have any suggestion so I guess it's not a big deal. We'll see when a third condition comes into play.